odoo/tools/cache.py

```diff
@@ -144,7 +144,7 @@ def determine_key(self):
         super(ormcache_multi, self).determine_key()
 
         # key_multi computes the extra element added to the key
-        spec = getargspec(self.method)
+        spec = getfullargspec(self.method)
```

Comment:
The result of this function is sent as parameter to **formatargspec** which is retro-compatible with **getfullargspec**
```python
def formatargspec(args, varargs=None, varkw=None, defaults=None,
                  kwonlyargs=(), kwonlydefaults={}, annotations={},
                  formatarg=str,
                  formatvarargs=lambda name: '*' + name,
                  formatvarkw=lambda name: '**' + name,
                  formatvalue=lambda value: '=' + repr(value),
                  formatreturns=lambda text: ' -> ' + text,
                  formatannotation=formatannotation):
    """Format an argument spec from the values returned by getfullargspec.

    The first seven arguments are (args, varargs, varkw, defaults,
    kwonlyargs, kwonlydefaults, annotations).  The other five arguments
    are the corresponding optional formatting functions that are called to
    turn names and values into strings.  The last argument is an optional
    function to format the sequence of arguments.

    Deprecated since Python 3.5: use the `signature` function and `Signature`
    objects.
    """
```